test/test_framework.py

```diff
@@ -161,22 +161,14 @@ def on_foo(self, event):
 
         framework.observe(pub.foo, obs.on_any)
         framework.observe(pub.bar, obs.on_any)
-        framework.observe(pub.foo, obs)  # Method name defaults to on_<event kind>.
 
-        try:
+        with self.assertRaises(RuntimeError):
             framework.observe(pub.baz, obs)
-        except RuntimeError as e:
-            self.assertEqual(
-                str(e),
-                'Observer method not provided explicitly'
-                ' and MyObserver type has no "on_baz" method')
-        else:
-            self.fail("RuntimeError not raised")
 
```

Comment:
+1 to using `self.assertRaises`, but note that you lost the error text assertion.

Reply:
yep! if it were important to catch in a unit test, I could do
```python
with self.assertRaises(RuntimeError, 'some funky message'):
    # ...
```

Reply:
You can do:

```
with self.assertRaises(SomeException) as cm:
    do_something()

the_exception = cm.exception
self.assertEqual(the_exception.error_code, 3)
```

Reply:
turns out the `msg` keyword argument to `assertRaises` does *nothing* (that I could see).
So instead I used `assertRaisesRegex`, that does exactly what i wanted.